core/field_image.js

```diff
@@ -81,6 +85,12 @@ Blockly.FieldImage.prototype.init = function() {
       'width': this.width_ + 'px'
     },
     this.fieldGroup_);
+
+    this.imageElement_.onclick = this.clickHandler_;
```

Comment:
Blockly does its own click and drag handling, because it makes occasionally complicated decisions about what is a click and what is a drag.  Instead of setting an onclick handler on an image, just define `Blockly.FieldImage.prototype.showEditor` to call this function if it's defined.

`showEditor` is called when Blockly determines that a gesture is a field click: https://github.com/google/blockly/blob/develop/core/gesture.js#L584